src/main/java/org/opensearch/knn/index/query/KNNQuery.java

```diff
@@ -23,26 +21,28 @@
 import org.opensearch.knn.index.KNNSettings;
 
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
 
 /**
  * Custom KNN query. Query is used for KNNEngine's that create their own custom segment files. These files need to be
  * loaded and queried in a custom manner throughout the query path.
  */
+@Getter
+@Builder
+@AllArgsConstructor
 public class KNNQuery extends Query {
 
     private final String field;
     private final float[] queryVector;
     private int k;
+    private Integer efSearch;
```

Comment:
It's good that we made Data Type Integer rather int , but are we checking null in efSearch?

Reply:
I believe so wherever required